src/plugins/ResizeReorder/ReorderHandle.js

```diff
@@ -0,0 +1,293 @@
+   */
+  onMouseMove = (deltaX) => {
+    this.cursorDeltaX += deltaX * (this.props.isRTL ? -1 : 1);
+  };
+
+  onMouseUp = () => {
+    this.props.toggleCellsRecycling(true, this.props.columnKey);
+    cancelAnimationFrame(this.frameId);
+    this.updateParentReorderingData({
+      isColumnReordering: false,
+      displacement: 0,
+    });
+    this.updateColumnOrder();
+    this.isReordering = false;
+    this.frameId = null;
+    this.cursorDeltaX = 0;
+    this.mouseMoveTracker.releaseMouseMoves();
+  };
+
+  /**
+   *
+   * @param {MouseEvent} event
+   */
+  initializeDOMMouseMoveTracker = (event) => {
+    this.mouseMoveTracker = new DOMMouseMoveTracker(
+      this.onMouseMove,
+      this.onMouseUp,
+      document.body,
+      this.props.touchEnabled
+    );
+    this.mouseMoveTracker.captureMouseMoves(event);
+  };
+
+  updateParentReorderingData = (state) => {
+    this.props.updateParentReorderingData(state);
+  };
+
+  updateDisplacementPeriodically = () => {
+    /* NOTE: We need to use requestAnimationFrame because whenever column reaches the end of table (scroll width is left),
+     we want to update the scrollX which can't be updated if we uer onMouseMove*/
+    this.frameId = requestAnimationFrame(this.updateDisplacementPeriodically);
+    this.updateDisplacementWithScroll();
+  };
+
+  /**
+   * @param {number} deltaX
+   * @return {number} deltaX bounded between cell group
+   */
+  getBoundedDeltaX = (deltaX) => {
+    // Column should not be moved beyond left of cell group
+    if (this.originalLeft + deltaX < 0) {
+      deltaX = -this.originalLeft;
+    }
+    // Column should not be moved beyond right of cell group
+    const cellGroupColumnWidths = this.props.getCellGroupWidth();
+    const cellGroupWidth = cellGroupColumnWidths.widths.reduce((a, b) => a + b);
+    const maxReachableDisplacement = cellGroupWidth - this.props.width;
+    if (this.originalLeft + deltaX > maxReachableDisplacement) {
+      deltaX = maxReachableDisplacement - this.originalLeft;
+    }
+    return deltaX;
+  };
+
+  updateDisplacementWithScroll = () => {
+    const scrollStart = this.scrollStart;
+    let { isFixed } = this.props;
+    let { scrollX, maxScrollX } = this.context;
+    let deltaX = this.cursorDeltaX;
+    if (!isFixed) {
+      // Relative dragX position on scroll
+      const dragX = this.originalLeft - scrollStart + deltaX;
+      const availableScrollWidth = this.props.availableScrollWidth;
+      deltaX += scrollX - scrollStart;
+
+      // Scroll the table left or right if we drag near the edges of the table
+      if (dragX > availableScrollWidth - DRAG_SCROLL_BUFFER) {
+        scrollX = Math.min(scrollX + DRAG_SCROLL_SPEED, maxScrollX);
+      } else if (dragX <= DRAG_SCROLL_BUFFER) {
+        scrollX = Math.max(scrollX - DRAG_SCROLL_SPEED, 0);
+      }
+      this.props.scrollToX(scrollX);
+    }
+    deltaX = this.getBoundedDeltaX(deltaX);
+    this.updateParentReorderingData({ displacement: deltaX });
+  };
```

Comment:
Update displacement wrt scroll position. It also:

Informs ResizeReorderCell to translate the div. (line 194)
Dispatches action to scroll to a specific position. (line 192)